### ISSUE TITLE
Qwen2/2.5/3 max_seq_len and EOS Masking

### ISSUE THREAD

**Issue by haydn-jones** [pytorch/torchtune#2792]
I've been trying to debug why I can't get my model to consistently stop its output, and it turns out that passing `max_seq_len` changes whether the EOS token is masked or not in SFT:
```python
from torchtune.data import InputOutputToMessages
from torchtune.models.qwen3 import qwen3_tokenizer

msg_fmt = InputOutputToMessages(
    masking_strategy="train_on_assistant",
)
inp = {
    "input": "A",
    "output": "B",
}

tokenizer = qwen3_tokenizer(
    path="./model/Qwen3-0.6B/vocab.json",
    merges_file="./model/Qwen3-0.6B/merges.txt",
)
out = tokenizer(msg_fmt(inp))
print(out["tokens"])
print(out["mask"])

# [151644, 872, 198, 32, 151645, 198, 151644, 77091, 198, 33, 151645]
# [True, True, True, True, True, True, False, False, False, False, False] <-- False

# Same tokenizer but set max_seq_len
tokenizer = qwen3_tokenizer(
    path="./model/Qwen3-0.6B/vocab.json",
    merges_file="./model/Qwen3-0.6B/merges.txt",
    max_seq_len=8192,
)
out = tokenizer(msg_fmt(inp))
print(out["tokens"])
print(out["mask"])

# [151644, 872, 198, 32, 151645, 198, 151644, 77091, 198, 33, 151645]
# [True, True, True, True, True, True, False, False, False, False, True]  <-- Now True
```

It gets flipped here: https://github.com/pytorch/torchtune/blob/c7a92e49935c40da088a9f6ae2b5b7b637f3f3a1/torchtune/data/_utils.py#L60

Because the Qwen2 tokenizer calls truncate with eos_id=True here:
https://github.com/pytorch/torchtune/blob/c7a92e49935c40da088a9f6ae2b5b7b637f3f3a1/torchtune/models/qwen2/_tokenizer.py#L416

This of course means that the SFT transform will use the cross entropy ignore ID as the target for the token before EOS rather than the EOS  ID. I assume this is not intended behavior?

**Comment by krammnic** [comment]
Hey @haydn-jones, I will definitely take a look, but we are planning to deprecate model-specific tokenizers, because we proposed a general one #2723. However, this problem is related to a truncation, so this is a good catch. Thanks for the issue!
